packages/gatsby/src/utils/validate-page-component.ts

```diff
@@ -0,0 +1,105 @@
+import path from "path"
+import fs from "fs-extra"
+import { IGatsbyPage } from "../redux/types"
+
+const validationCache = new Map<string, boolean>()
```

Comment:
A set seems more appropriate.

```suggestion
const validationCache = new Set<string>()
```